processing/src/main/java/org/apache/druid/query/aggregation/BufferAggregator.java

```diff
@@ -72,7 +72,9 @@
    *
    * Converts the given byte buffer representation into an intermediate aggregate Object
    *
-   * <b>Implementations must not change the position, limit or mark of the given buffer</b>
+   * <b>Implementations must not change the position, limit or mark of the given buffer.</b>
+   *
+   * <b>The object returned must not have any references to the given buffer (i.e., make a copy).</b>
```

Comment:
I'd include rationale for this: it's because the object returned by `get` may live beyond the lifetime of the underlying buffer.

Reply:
Added the rationale here.